src/pages/ProfilesExplorerView/components/SceneExploreServiceFlameGraph/components/SceneFunctionDetailsPanel/components/GitHubContextProvider/infrastructure/GitSessionCookieManager.ts

```diff
@@ -48,15 +54,18 @@ class InternalGitSessionCookieManager implements GitSessionCookieManager {
   }
 
   private syncCookieWithBrowser(): void {
-    const cookie = InternalGitSessionCookieManager.getCookieFromJar(document.cookie, GITHUB_SESSION_COOKIE_NAME);
+    const cookie = InternalGitSessionCookieManager.getCookieFromJar(
+      document.cookie,
+      ACCEPTED_GITHUB_SESSION_COOKIE_NAMES
+    );
     if (cookie?.key === this.rawCookie?.key && cookie?.value === this.rawCookie?.value) {
       return;
     }
 
     cookie !== undefined ? this.setCookie(`${cookie.key}=${cookie.value}`) : this.deleteCookie();
   }
 
-  private static getCookieFromJar(jar: string, name: string): Cookie | undefined {
+  private static getCookieFromJar(jar: string, candidates: string[]): Cookie | undefined {
```

Comment:
I think this method should do a full scan of the cookie jar for each candidate. It should return the cookie of the first possible match. This way we'll always use cookies of candidates first in list before candidates later in the list. For example:

```
candidates = [ 'pyroscope_git_session', 'GitSession' ]
```

First, we want to check if there's any `pyroscope_git_session` cookie in the jar. Even if we find a `GitSession` first, we should keep checking if there's a `pyroscope_git_session` later. If we don't find one, we can do another scan looking for `GitSession`.

With this current implementation, we'll accept the first of either `GitSession` or `pyroscope_git_session`, which might not be useful when we're trying to migrate to using just `pyroscope_git_session`.